quickwit/quickwit-ingest/src/ingest_v2/workbench.rs

```diff
@@ -29,14 +32,17 @@ use tracing::warn;
 
 /// A helper struct for managing the state of the subrequests of an ingest request during multiple
 /// persist attempts.
+#[derive(Default)]
 pub(super) struct IngestWorkbench {
-    subworkbenches: HashMap<SubrequestId, IngestSubworkbench>,
-    num_successes: usize,
+    pub subworkbenches: HashMap<SubrequestId, IngestSubworkbench>,
+    pub num_successes: usize,
+    pub num_attempts: usize,
+    pub max_num_attempts: usize,
```

Comment:
Can we add a comment explaining this si an overall number of attempts. Not a per sub request number.